inst/tests/tests.Rraw

```diff
@@ -9877,6 +9877,13 @@ test(1757, fread("winallquoted.csv")[c(1,2,4998,4999)],
 
 test(1758, sapply(fread("A,B\n,"),class), c(A="logical",B="logical"))
 
+# split.data.table should respect non-alphabetic order if passed a factor in by, #2082
+DT = data.table(a = factor(c('a', 'b'), levels = c('b', 'a')))
+split(DT, by = 'a', sorted = TRUE)
+test(1759, split(DT, by = 'a', sorted = TRUE),
```

Comment:
Another tests with more than one column to check this fix works in combination with other columns would be great please.